File: 백준/Silver/1436. 영화감독 숌/영화감독 숌.py
# 666부터 검사 (그 전에는 종말의 수 안나옴)
# i를 1씩 증가시키면서 str(i) 안에 '666'이 있는지 체크
# 있으면 카운트 증가하고, 카운트가 목표 달성했는지 체크

N = int(input())

i = 666
count = 0
while True:
    if '666' in str(i):
        count += 1
    if count == N:
        print(i)
        break
    i += 1

File: 백준/Silver/1436. 영화감독 숌/영화감독 숌.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int n = Integer.parseInt(br.readLine());

		// 완탐
		int count = 1;
		int num = 666; // 첫 수부터 시작 (n >= 1)

		while (count < n) {
			num++;

			String temp = String.valueOf(num);
			if (temp.contains("666")) {
				count++;
			}
		}

		System.out.println(num);
	}
}